functions/src/controllers/locations.js

```diff
@@ -0,0 +1,146 @@
+                country,
+                createdAt: new Date().toISOString(),
+                image,
+                lga,
+                name,
+                price,
+                state,
+                status,
+                trafficRate,
+            };
+            const location = await db.collection('locations').doc().create(data).then(ref => ref);
+            return res.status(CREATED).send({
+                data: { location, message: 'Location successfully created' },
+                status: 'success',
+            });
+        } catch (error) {
+            return res.status(INTERNAL_SERVER_ERROR).send({                
+                message: getStatusText(INTERNAL_SERVER_ERROR),
+                status: 'error',
+            });
+        }
+    },
+
+    async deleteLocation(req, res) {
+        try {
+            const location = await db.collection('locations').doc(req.params.id);
+            await location.delete();
+            if (!location) {
+                return res.status(NOT_FOUND).send({
+                    message: getStatusText(NOT_FOUND),
+                    status: 'error',
+                });
+            }
+            return res.status(OK).send({
+                data: null,
+                status: 'success',
+            });
+        } catch (error) {
+            return res.status(INTERNAL_SERVER_ERROR).send({
+                error,
+                message: getStatusText(INTERNAL_SERVER_ERROR),
+                status: 'error',
+            });
+        }
+    },
+
+    async getAll(req, res) {
+        try {
+            const query = await db.collection('locations');
+            const locations = [];
+            await query.get().then(querySnapshot => {
+                const docs = querySnapshot.docs;
+                for (const doc of docs) {
+                    const selectedItem = {
+                        id: doc.id,
+                        location: doc.data(),
+                    };
+                    locations.push(selectedItem);
+                }});
+            return res.status(OK).send({
+                data: { locations },
+                status: 'success',
+            });
+        } catch (error) {
+            return res.status(INTERNAL_SERVER_ERROR).send({
+                error,
+                message: getStatusText(INTERNAL_SERVER_ERROR),
+                status: 'error',
+            });
+        }
+    },
+
+    async getOne(req, res) {
+        try {
+            const document = db.collection('locations').doc(req.params.id);
+            if (!document) {
+                return res.status(NOT_FOUND).send({
+                    message: getStatusText(NOT_FOUND),
+                    status: 'error',
+                });
+            }
+            const documentData = await document.get();
+            const location = documentData.data();
+            return res.status(OK).send({
+                data: { location },
+                status: 'success',
```

Comment:
Single location should be an object, multiple locations an array
return data: location, or location as location